lms/static/scripts/frontend_apps/utils/url.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Polyfill for `Object.hasOwn`.
+ *
+ * `hasOwn(someObject, property)` should be used instead of
+ * `someObject.hasOwnProperty(name)`.
+ */
+function hasOwn(object: object, property: string): boolean {
+  return Object.prototype.hasOwnProperty.call(object, property);
+}
+
+/**
+ * Replace parameters in a URL template with values from a `params` object.
+ *
+ * Returns an object containing the expanded URL and a dictionary of unused
+ * parameters.
+ *
+ *   replaceURLParams('/things/:id', {id: 'foo', q: 'bar'}) =>
+ *     {url: '/things/foo', unusedParams: {q: 'bar'}}
+ */
+export function replaceURLParams<Param>(
+  urlTemplate: string,
+  params: Record<string, Param>,
+): { url: string; unusedParams: Record<string, Param> } {
```

Comment:
If we don't have a use case for allowing unused parameters here, then we could change the behavior to _disallow_ them (ie. throw if there are any) and return just the URL.